daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala

```diff
@@ -1114,60 +1114,61 @@ private[lf] object Speedy {
   private[speedy] sealed abstract class Kont {
 
     /** Execute the continuation. */
-    def execute(v: SValue): Control
+    def execute(machine: Machine, v: SValue): Control
   }
 
   /** Final continuation; machine has computed final value */
   private[speedy] final case object KFinished extends Kont {
-    override def execute(v: SValue): Control = {
-      Control.Complete(v)
-    }
+    override def execute(machine: Machine, v: SValue): Control = Control.Complete(v)
   }
 
-  private[speedy] final case class KOverApp(machine: Machine, newArgs: Array[SExprAtomic])
-      extends Kont
+  private[speedy] final case class KOverApp(
```

Comment:
Again, I am bit confused to the motivation of this change.

Is it just a preferred style to have the `object` `apply` method access the `machine`  components?

Or is there a more fundamental reason to avoid passing the full `machine` to the `KOverApp` construction?

Reply:
I did not want to change the calling sites, to shows there is no big impact how you use it.
I see the additional advantage, that if we need to add in a continuation more data from the machine we can do it locally without modifying the calling site.

But this is more cosmetic, I am fine with inlining the body of apply where it is call.

Reply:
no, it is fine as it is

Reply:
With the use cases where we introduce a companion object with an apply, am I correct in understanding that the intent is to only use the apply to contract case class instances? If so, perhaps we should make the case class constructors private?

Reply:
abcac65f194c2202b72d83d715eadfcd0f143d7e